beacon_node/http_api/src/publish_blocks.rs

```diff
@@ -114,12 +116,31 @@ pub async fn publish_block<T: BeaconChainTypes, B: IntoGossipVerifiedBlockConten
     let blobs_opt = block_contents.inner_blobs();
 
     /* if we can form a `GossipVerifiedBlock`, we've passed our basic gossip checks */
-    let (gossip_verified_block, gossip_verified_blobs) = block_contents
-        .into_gossip_verified_block(&chain)
-        .map_err(|e| {
-            warn!(log, "Not publishing block, not gossip verified"; "slot" => slot, "error" => ?e);
-            warp_utils::reject::custom_bad_request(e.to_string())
-        })?;
+    let (gossip_verified_block, gossip_verified_blobs) =
+        match block_contents.into_gossip_verified_block(&chain) {
+            Ok(b) => b,
+            Err(BlockContentsError::BlockError(BlockError::BlockIsAlreadyKnown)) => {
```

Comment:
@michaelsproul mentioned [in this comment](https://github.com/sigp/lighthouse/pull/4655#discussion_r1303856469) that it's meant to cover this scenario:
> One of the relays also hit the error with full blocks when broadcasting to multiple BNs, and it can happen for VCs that broadcast too (e.g. Vouch).

so I guess the same would apply to blobs too - if we already observed the corresponding blob sidecar (before the block arrives) then it would be considered a duplicate proposal?

Reply:
My brain has too many other things going on at the moment to think about this properly, so I just opened an issue: https://github.com/sigp/lighthouse/issues/4725. I think we should merge this PR and come back to it.